### PR TITLE
Task/iotam autoregistration

### LINKED ISSUE #124: Implement Autoregistration in IOTA Manager
Implement the autoregistration of the IOTA in a IOTA Manager, on start.

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #124

### CODE REVIEW

File: test/unit/iotamRequests/registrationEmpty.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "protocol": "GENERIC_PROTOCOL",
+  "description": "A generic protocol",
+  "iotagent": "http://smartGondor.com",
+  "resource": "/iot/d",
+  "services": []
+}
```
Comment (gtorodelvalle) [inline test/unit/iotamRequests/registrationEmpty.json]: Final blank line appreciated... :)

File: test/unit/iotamRequests/registrationWithGroups.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    {
+      "apikey": "801230BJKL23Y9090DSFL123HJK09H324HV8732",
+      "token": "8970A9078A803H3BL98PINEQRW8342HBAMS",
+      "entity_type": "SensorMachine",
+      "resource": "/deviceTest",
+      "service": "theService",
+      "service_path": "theSubService"
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/iotamRequests/registrationWithGroups.json]: Final blank line appreciated... :)

File: test/unit/iotamResponses/registrationSuccess.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (gtorodelvalle) [inline test/unit/iotamResponses/registrationSuccess.json]: Final blank line appreciated... :)

**Comment by gtorodelvalle** [comment]
Still warming up after holidays :p but looking great to me apart from 3 minor comments... :) LGTM as soon as they are fixed... ;)

**Comment by dmoranj** [comment]
All blank lines fixed in 45fb324

**Comment by gtorodelvalle** [comment]
LGTM ;) Thanks @dmoranj :)

**Pull request merged**